issues: implement functionality to add images to post

FaiyazMujawar opened the pull request:
Fixes #51 

In this PR,

- Implemented a separate route for posting images.
- Implemented an Image Mongoose Schema, for storing each image meta data for logging purpose.
- Implemented functionality to add images to post.
- Updated documentation accordingly.

drigio commented on controllers/IssueController.js:
```diff
@@ -398,6 +442,7 @@ const validateData = (title, description, section, scope) => {
       .regex(/\w+/)
       .rule({ message: "Please provide a valid description" })
       .required(),
+    images: Joi.array().max(3).required(),
```
Images arent `required` right?

FaiyazMujawar replied:
They are required. Even if there are jo images, an empty array must be passed to maintain consistency in database schema

drigio replied:
Okay

drigio commented on models/Issue.js:
```diff
@@ -10,6 +10,10 @@ const IssueSchema = new Schema({
   },
+  images: {
+    type: [String],
+    required: true,
```
Do we have to make images `required`?

drigio commented on services/FileService.js:
```diff
@@ -0,0 +1,77 @@
+const storage = multer.diskStorage({
+  destination: IMAGE_DESTINATION,
+  filename: (_, { originalname }, cb) => {
+    cb(null, `${originalname}${extname(originalname)}`);
```
Do you think we should be naming it this way? Cuz if there are multiple images with the same name, they might get overridden, which can be a possibility during multiple requests.

FaiyazMujawar replied:
It was just to time being. As u earlier said, it would be hashed

drigio replied:
Make that change then before we merge in.

drigio commented on services/FileService.js:
```diff
@@ -0,0 +1,77 @@
+    }
+    return null;
+  } catch (error) {
+    throw error;
```
throwing error, if not handled later might crash the system. Are you catching it at a later stage?

drigio commented on controllers/IssueController.js:
```diff
@@ -130,18 +171,21 @@ const addIssue = async (req, res) => {
         id
       );
+      console.log("issue created");
+      await updateImageIssueId(images, id);
```
`images` can be null if no image is uploaded for an issue and it might give errors.

FaiyazMujawar replied:
Nope. $in would not work if there are no matches. Hence I made images as required.

drigio replied:
If there are no images we shouldn't call this method in the first place.

FaiyazMujawar replied:
Okay. Can be checked at the time of issue creation.

drigio reviewed (approved):
LGTM! I'll merge in once you give me a head's up. Also I hope you have seen the new code working?

FaiyazMujawar committed:
issues: implement functionality to add images to post

In this PR,

- Implemented a separate route for posting images.
- Implemented an Image Mongoose Schema, for storing each image meta data for logging purpose.
- Implemented functionality to add images to post.
- Updated documentation accordingly.
